### DISCUSSION TITLE
Comms notifications oddities

### DISCUSSION THREAD

**Discussion by Deadweasel** [iaincollins/icarus#14] [Bugs & Problems]
Still loving this little app so far!

A couple of things I've noticed about the comms popup notifications lately:
1) They linger.  Like, for a loooong time, enough so that in a busy area, there's enough NPC chatter to keep the thing scrolling and blocking off the right 1/3 of the screen.
2) They actually show messages sent back and forth by live players in a system, even if I'm playing in Solo mode!  It's actually kind of neat, feeling like a ghost in the station, where they're yammering on about almost colliding with traffic, and I see none of it, hehehe

Those are the only things I've had any issues with.  Otherwise, this app is freaking SOLID.

**Comment by iaincollins** [comment]
Thanks for the bug reports!

> 1. They linger.  Like, for a loooong time, enough so that in a busy area, there's enough NPC chatter to keep the thing scrolling and blocking off the right 1/3 of the screen.

Oh that's weird, they shouldn't last too long - only for about the same time chat messages linger on screen - this is about 8 seconds by default (intended to be just long enough to notice and read a longer message), but that may seem quite long if the chat is busy.

If they are being annoying and getting in the way, there is a little icon in the menu at the top that looks like a chat message bubble that you can click to mute them for the window (it will say "notifications on" or "notifications off" and the icon will toggle state when you click/tap it). This is per-window and will stay that way until you change it or close the window.

In the longer term I'd like to improve the UX, and make it possible to select what sort of notifications you see and to view old messages (e.g. chat history) but the option to mute them entirely on a per-window basis is an interim compromise.
 
> 2. They actually show messages sent back and forth by live players in a system, even if I'm playing in Solo mode!  It's actually kind of neat, feeling like a ghost in the station, where they're yammering on about almost colliding with traffic, and I see none of it, hehehe

Oh yeah! So this is weird right? It's actually a bug in the game! It's been happening for a few months now and happens in the in-game chat as well, super odd!

I can't do anything about this one but it would be nice if they (a) fix or (b) [ even better? ] expanded the in-game chat to allow opt-in galaxy wide public channels. I can see the option of an in-game IRC (text chat) client being appealing for a bunch of players, especially given how hard it is to run into players in such a huge environment [outside of major hubs].

**Reply by Deadweasel** [reply]
The chat bug makes sense, but it's weird that I saw it on Icarus and never noticed it in-game before.  Guess that just further proves this app's utility! :D

About the long-term notifications, they seem to be most frequently associated with NPC dialog directed at me, usually ships sent after me related to a mission.  I just had a pair of "chats" from a NPC that wouldn't seem to go away at all until I actually refreshed the browser, even after I turned off the notifications altogether.  Not a huge thing or even close to a game-breaker for me, just wondered why those particular notifications would stand out and linger.

I have yet to prove this effect on a different system, so it could be platform-related.  So far I have only caught it happening on Chromium, running on Raspbian OS, Bullseye distro (the Pi4 drives a touch screen for Icarus to strut its stuff).  I'll update if I can narrow it down to that or observe it happening on the Windows browser/host terminal window side.

**Reply by iaincollins** [reply]
Thanks for the that detail! That does sound annoying.

I think that's a bug in the library I'm using for notifications. It's actually the only part of the UI driven by a third party library (as notification behavior is notoriously tricky to get right) but it seems it has some known issues. I've seen a few of them myself - and implemented workarounds for them in the code where possible - but not seen that issue specifically.

I've put in a workaround for this issue in the next update. Mow when muting notifications after 2-3 seconds it will hide *all* notifications until you toggle them visible again. This doesn't fix the issue exactly, in that stuck notifications will still linger and show up when you unmute notifications, but at least the toggle to hide them will always, definitely hide them.

I might refine this a bit further as a system to periodically check for and purge stuck messages would be nice, but might just be better to swap out the notification library at that point (even if that means writing a custom one).

Either way the change above will be out in the next update, should be sometime this week!

**Reply by iaincollins** [reply]
^ This update has now gone out in v0.13.2
